activerecord/test/cases/insert_all_test.rb

```diff
@@ -136,4 +136,26 @@ def test_upsert_all_does_not_perform_an_upsert_if_a_partial_index_doesnt_apply
 
     assert_equal ["Out of the Silent Planet", "Perelandra"], Book.where(isbn: "1974522598").order(:name).pluck(:name)
   end
+
+  def test_insert_all_defaults
+    skip unless supports_insert_on_duplicate_skip?
+
+    defaults = { published_on: Date.new(1938, 4, 1) }
+    Book.insert_all([{ name: "Out of the Silent Planet", author_id: 7, isbn: "1974522598" }], defaults: defaults)
+
+    book = Book.find_by(name: "Out of the Silent Planet")
+    assert_equal defaults[:published_on], book.published_on
+  end
+
+  def test_insert_all_defaults_not_used_when_specified
+    skip unless supports_insert_on_duplicate_skip?
+
+    defaults = { published_on: Date.new(1938, 4, 1) }
+    published_on = Date.new(1940, 4, 1)
+    Book.insert_all([{ name: "Out of the Silent Planet", author_id: 7, isbn: "1974522598", published_on: published_on }], defaults: defaults)
+
+    book = Book.find_by(name: "Out of the Silent Planet")
+    assert_not_equal defaults[:published_on], book.published_on
+    assert_equal published_on, book.published_on
```

Comment:
These tests, too, don't need 

```ruby
 skip unless supports_insert_on_duplicate_skip?
```

Just change `Book.insert_all` to `Book.insert_all!`

(since the new functionality here doesn't require the conflict handling part — just the bulk insert part)